python/cuml/neighbors/nearest_neighbors.pyx

```diff
@@ -204,16 +204,66 @@ class NearestNeighbors(Base):
         self.n_neighbors = n_neighbors
         self._should_downcast = should_downcast
         self.n_indices = 0
+        self.sizes = None
+        self.input = None
 
     def __del__(self):
 
         # Explicitly free these since they were allocated
         # on the heap.
         if self.n_indices > 0:
-            free(<int*><size_t>self.sizes)
-            free(<float**><size_t>self.input)
+            if self.sizes is not None:
+                free(<int*><size_t>self.sizes)
+            if self.input is not None:
+                free(<float**><size_t>self.input)
             self.n_indices = 0
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+
+        if self.n_indices > 1:
```

Comment:
should this be `self.n_gpus`?

Reply:
This is intended to be `n_indices` here. While `n_gpus` is only used in the case of non-dask SPMG memory chunking, the `n_indices` is used throughout the entire object. This is another artifact of our separating our core cuML API from the distributed API and attempting to condense single-GPU and multi-GPU into the same class. Once OPG is our standard and we unify our core cuML & Dask APIs, this separation between `n_gpus` and `n_indices` will go away.